From c598a89dfa7771ed738b6e66b9e56e59196763b0 Mon Sep 17 00:00:00 2001
From: Adrian-Ken Rueegsegger <ken@codelabs.ch>
Date: Thu, 5 Oct 2017 18:08:07 +0200
Subject: [PATCH] static_mmio: Disable SPARK for Set_Base_Address body

Otherwise a warning will be issued that the procedure has no effect.
However, we need to implement the subprogram since it is specified in
the common HW.MMIO_Range spec.
---
 ada/static_mmio/hw-mmio_range.adb | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/ada/static_mmio/hw-mmio_range.adb b/ada/static_mmio/hw-mmio_range.adb
index 7e111a7..be5e41e 100644
--- a/ada/static_mmio/hw-mmio_range.adb
+++ b/ada/static_mmio/hw-mmio_range.adb
@@ -31,6 +31,8 @@ is
       MMIO (Index) := Value;
    end Write;
 
-   procedure Set_Base_Address (Base : Word64) is null;
+   procedure Set_Base_Address (Base : Word64)
+   is null
+   with SPARK_Mode => Off;
 
 end HW.MMIO_Range;
-- 
2.1.4

